### PR TITLE
C++11 Absolute Value

### PR DESCRIPTION
**Pull request by varunagrawal**
Replaced `fabs` with `std::abs` to be more C++11 compliant.

I have used `std::abs` everywhere instead of simply `abs` since I didn't want accidental linking against the C version of `abs` which only works for integers. Explicit is better than implicit.

https://stackoverflow.com/questions/33738509/whats-the-difference-between-abs-and-fabs

---
This change is [Reviewable](https://reviewable.io/reviews/borglab/gtsam/115)

### CODE REVIEW

**Comment by varunagrawal** [comment]
@dellaert you may want to reconcile the `bugfix/Huber` branch which was last looked at in 2016.

**Comment by dellaert** [comment]
On your recommendation :-) will you also create an issue to track this?

**Comment by dellaert** [comment]
I think this should be split into two PRs. One is easy (std::abs). Other one should correspond to issue about Huber.

**Comment by varunagrawal** [comment]
@dellaert this PR now only deals with the absolute value computation.

File: gtsam/linear/tests/testNoiseModel.cpp [hunk 1]
```diff
@@ -452,19 +452,24 @@ TEST(NoiseModel, WhitenInPlace)
 /*
  * These tests are responsible for testing the weight functions for the m-estimators in GTSAM.
  * The weight function is related to the analytic derivative of the residual function. See
- *  http://research.microsoft.com/en-us/um/people/zhang/INRIA/Publis/Tutorial-Estim/node24.html
+ *  https://members.loria.fr/MOBerger/Enseignement/Master2/Documents/ZhangIVC-97-01.pdf
```
Comment (dellaert) [inline gtsam/linear/tests/testNoiseModel.cpp]: Changes in this file (except std::abs) belong in other PR

Comment (varunagrawal) [reply gtsam/linear/tests/testNoiseModel.cpp]: I just force-pushed so the PRs should now be truly separate.

**Pull request merged**